src/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -1,33 +1,68 @@
 import React, { useRef, forwardRef, useCallback, useMemo, useEffect, useState } from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
-import Button from "../../components/Button/Button";
+import Button from "../Button/Button";
 import usePrevious from "../../hooks/usePrevious";
 import useMergeRefs from "../../hooks/useMergeRefs";
 import { backwardCompatibilityForProperties } from "../../helpers/backwardCompatibilityForProperties";
 import { baseClassName } from "./ButtonGroupConstants";
 import { ButtonWrapper } from "./ButtonWrapper";
 import "./ButtonGroup.scss";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import { BASE_SIZES, SIZES } from "../../constants/sizes";
+import { ButtonType, Size } from "../Button/ButtonConstants";
 
-const ButtonGroup = forwardRef(
+type ButtonGroupOption = {
+  icon?: string | React.FunctionComponent | null;
+  leftIcon?: string | React.FunctionComponent | null;
+  ariaLabel?: string;
+  subText?: string;
+  value: number;
+  text: string;
+  disabled?: boolean;
+  tooltipContent?: string;
+};
+
+interface ButtonGroupProps extends VibeComponentProps {
+  className?: string;
+  componentClassName?: string;
+  options: ButtonGroupOption[];
+  value: string | number;
+  onSelect: (value: number, name: string) => void;
+  size?: Size;
+  kind?: ButtonType.SECONDARY | ButtonType.TERTIARY;
+  name?: string;
+  disabled?: boolean;
+  groupAriaLabel?: string;
+  tooltipPosition?: string;
+  tooltipHideDelay?: number;
+  tooltipShowDelay?: number;
+  tooltipContainerSelector?: string;
+  tooltipMoveBy?: { main: number; secondary: number };
+}
+
+const ButtonGroup: React.ForwardRefExoticComponent<ButtonGroupProps & React.PropsWithChildren<unknown>> & {
+  sizes?: typeof SIZES;
+  kinds?: typeof ButtonType;
+} = forwardRef(
   (
     {
-      className,
+      className = undefined,
       // Backward compatibility for props naming
       componentClassName,
       options,
-      name,
-      disabled,
-      value,
-      onSelect,
-      size,
-      kind,
-      groupAriaLabel,
-      tooltipPosition,
-      tooltipHideDelay,
-      tooltipShowDelay,
-      tooltipContainerSelector,
-      tooltipMoveBy
+      name = "",
+      disabled = false,
+      value = "",
+      onSelect = undefined,
+      size = BASE_SIZES.SMALL,
+      kind = ButtonType.SECONDARY,
+      groupAriaLabel = "",
+      tooltipPosition = undefined,
+      tooltipHideDelay = undefined,
+      tooltipShowDelay = undefined,
+      tooltipContainerSelector = undefined,
+      tooltipMoveBy = undefined
```

Comment:
No need to specify `undefined` as default value, but not a blocker at all

Reply:
Noted. Removed the default undefined props in the component.